core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,48 @@
+{% macro get_show_grant_sql(relation) %}
+{{ return(adapter.dispatch("get_show_grant_sql", "dbt")(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+show grants on {{ relation.type }} {{ relation }}
+{% endmacro %}
```

Comment:
Nice! What information do we need this macro to return?  I think it's just:
- privileges being granted (`select`, `usage`, etc)
- recipients (also called "principals" or "grantees")

That is, for _this_ relation, excluding privileges held by the current user/role/principal. I'm also realizing, we should probably limit to just the grants given _by_ this user/role/principal, to guarantee that any subsequent `revoke` statements will succeed.

As previously mentioned, `show grants` feels like a good standard/default. Assuming we're starting by testing with Postgres, we'll need to implement a PostgreSQL version, which looks a little different:
```sql
{% macro postgres__get_show_grant_sql(relation) %}

    select
    
        -- we'll need to standardize these names across databases
        -- which ones do we want to pick?
        
        grantee,          -- or 'recipient'? or 'principal'?
        privilege_type    -- or 'privilege'?
    
    from information_schema.role_table_grants
    where grantor = current_role
      and grantee != current_role
      and table_schema = '{{ relation.schema }}'
      and table_name = '{{ relation.identifier }}'
        
{% endmacro %}
```